File: android/src/main/java/br/dev/michaellopes/flutter_anycam/camera/CameraBridge.java
package br.dev.michaellopes.flutter_anycam.camera;

import java.util.HashMap;
import java.util.Map;

import br.dev.michaellopes.flutter_anycam.integration.FlutterEventChannel;

public class CameraBridge {
    final int viewId;

    public CameraBridge(int viewId) {
        this.viewId = viewId;
    }


    public void onConnected(Map<String, Object> data) {
        FlutterEventChannel.getInstance().send(viewId, "onConnected", data);
    }

    public void onDisconnected() {
        FlutterEventChannel.getInstance().send(viewId, "onDisconnected", new HashMap());
    }

    public void onUnauthorized() {
        FlutterEventChannel.getInstance().send(viewId, "onUnauthorized", new HashMap());
    }

    public void onFailed(String message) {
        FlutterEventChannel.getInstance().send(viewId, "onFailed", new HashMap() {{
            put("message", message);
        }});
    }

    public void  onVideoFrameReceived(Map<String, Object> imageData) {
        FlutterEventChannel.getInstance().send(viewId, "onVideoFrameReceived", imageData);
    }
}

File: ios/Classes/Camera/CameraBridge.swift
//
//  CameraBridge.swift
//  Pods
//
//  Created by Michael Lopes on 28/09/25.
//

class CameraBridge {
    let viewId: Int;
    
    private var isConnectedCalled = false;
    
    init(viewId: Int) {
        self.viewId = viewId;
    }
    
    func onConnected(data: [String: Any?]) -> Void {
        isConnectedCalled = true;
        FlutterEventStreamChannel.shared.send(self.viewId, "onConnected", data);
    }
    
    func onDisconnected() -> Void {
        FlutterEventStreamChannel.shared.send(self.viewId, "onDisconnected", [String : Any?]());
    }
    
    func onUnauthorized() -> Void {
        FlutterEventStreamChannel.shared.send(self.viewId, "onUnauthorized", [String : Any?]());
    }
    
    func onFailed(message: String) -> Void {
        FlutterEventStreamChannel.shared.send(self.viewId, "onFailed",  [
            "message": message
        ])
    }
    
    func onVideoFrameReceived(imageData: [String: Any]) -> Void {
        if(isConnectedCalled) {
            FlutterEventStreamChannel.shared.send(self.viewId, "onVideoFrameReceived", imageData)
        }
    }
}
